org-jira.el

```diff
@@ -1027,20 +1079,22 @@ Expects input in format such as: [2017-04-05 Wed 01:00]--[2017-04-05 Wed 01:46]
           (> (time-to-seconds (date-to-time (car a)))
              (time-to-seconds (date-to-time (car b)))))))
 
-(defun org-jira-update-worklogs-for-current-issue ()
-  "Update the worklogs for the current issue."
-  (lexical-let ((issue-id (org-jira-get-from-org 'issue 'key)))
-    ;; Run the call
-    (jiralib-get-worklogs
-     issue-id
-     (cl-function
-      (lambda (&rest data &allow-other-keys)
-        (ensure-on-issue-id
-         issue-id
-         (let ((worklogs (org-jira-find-value (cl-getf data :data) 'worklogs)))
-           (org-jira-logbook-reset
-            issue-id
-            (org-jira-sort-org-clocks (org-jira-worklogs-to-org-clocks worklogs))))))))))
+(defun org-jira-update-worklogs-for-current-issue (&optional force)
+  "Update the worklogs for the current issue when `org-jira-worklog-sync-p' is not nil. 
+You can `force' the update by passing a non-nil argument."
+  (when (or force org-jira-worklog-sync-p)
```

Comment:
Same concern as other comment - more functions that are useless under certain state conditions and now being nested in an entire conditional for a function.